src/app/Scenes/MyCollection/MyCollection.tsx

```diff
@@ -171,6 +171,10 @@ const MyCollection: React.FC<{
     }
   }, [artworks.length, filtersCount, isSearchBarVisible])
 
+  useEffect(() => {
+    reInitializeLocalArtworkFilter(artworks)
+  }, [artworks.length])
```

Comment:
Just checking for the number of artworks wouldn't update the filters in a couple of cases. `artworks.length` would stay the same if an artwork gets updated.

```suggestion
  }, [artworks])
```

Reply:
True that! Thanks.